module/renderer/src/commonMain/kotlin/org/cru/godtools/shared/renderer/content/RenderTabs.kt

```diff
@@ -0,0 +1,129 @@
+import org.cru.godtools.shared.renderer.state.State
+import org.cru.godtools.shared.tool.parser.model.Tabs
+import org.cru.godtools.shared.tool.parser.model.primaryColor
+import org.cru.godtools.shared.tool.parser.model.primaryTextColor
+import org.cru.godtools.shared.tool.parser.model.stylesParent
+
+private val CORNER_RADIUS = 10.dp
+
+internal const val TestTagTabs = "tabs"
+internal const val TestTagTab = "tab"
+internal val TabIsSelected = SemanticsPropertyKey<Boolean>(
+    name = "TabIsSelected",
+    mergePolicy = { parentValue, _ ->
+        // Never merge TestTags, to avoid leaking internal test tags to parents.
+        parentValue
+    }
+)
+
+@Composable
+internal fun ColumnScope.RenderTabs(tabs: Tabs, state: State, modifier: Modifier = Modifier) {
+    val invisible by remember(tabs, state) {
+        tabs.isInvisibleFlow(state)
+    }.collectAsState(tabs.isInvisible(state))
+
+    var selectedIndex by remember { mutableIntStateOf(0) }
+
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val borderColor = tabs.stylesParent.primaryColor.toComposeColor()
+    val selectedTab: Tabs.Tab? = tabs.tabs.getOrNull(selectedIndex)
+
+    LaunchedEffect(tabs, state) {
+        // handle play/stop listeners
+        state.contentEvents
+            .flowWithLifecycle(lifecycleOwner.lifecycle, Lifecycle.State.RESUMED)
+            .collect { event ->
+                tabs.tabs.firstOrNull { event in it.listeners }?.let {
+                    selectedIndex = it.position
+                }
+            }
+    }
+
+    SecondaryTabRow(
+        selectedTabIndex = selectedIndex,
+        modifier = modifier
+            .testTag(tag = TestTagTabs)
+            .visibility(model = tabs, state = state)
+            .padding(horizontal = HorizontalPadding)
+            .border(
+                BorderStroke(width = 2.dp, borderColor),
+                RoundedCornerShape(CORNER_RADIUS)
+            )
+            // Needed to clip SecondaryTabRow when applying border modifier. ~Levi
+            .clip(RoundedCornerShape(CORNER_RADIUS)),
+        containerColor = Color.White,
+        contentColor = Color.White,
+        indicator = @Composable {
+        },
+        divider = @Composable {
+        },
+        tabs = @Composable {
+            tabs.tabs.forEachIndexed { index, tab ->
+
+                val isSelected: Boolean = index == selectedIndex
+                val selectedColor = tab.stylesParent.primaryColor.toComposeColor()
+                val unselectedColor = tab.stylesParent.primaryTextColor.toComposeColor()
+                val backgroundColor = if (isSelected) selectedColor else unselectedColor
+
+                Tab(
+                    selected = isSelected,
+                    onClick = {
+                        selectedIndex = index
+                    },
+                    modifier = Modifier
+                        .testTag(TestTagTab)
+                        .semantics { set(TabIsSelected, isSelected) }
```

Comment:
I am curious if the selected state is present in the semantics tree already since it is set directly on the tab, and it probably needs to be present for accessibility.

Reply:
if not, then this works great to expose it for tests